### ISSUE TITLE
BAMPE mode: only properly paired reads were used for peak calling ?

### ISSUE THREAD

**Issue by carl-** [taoliu/MACS#134]
HI tao,

It seems that if only properly paired read were used in BAMPE mode. Is that true ? I just want to confirm that .

**Comment by taoliu** [comment]
Yes. It requires FLAG 0x2: [MACS2/IO/Parser.pyx#L1147](https://github.com/taoliu/MACS/blob/master/MACS2/IO/Parser.pyx#L1147)

**Comment by carl-** [comment]
thanks. I read the BAMPE parser section in Parser.pyx .
an extra intend is need for each line from 1155 to 1162, in case of someone (such as me ) merges se end and pe end read to a single bam and then feed it to macs2 callpeak in bampe mode.

**Comment by carl-** [comment]
Is there any possible to support single and and pair end reads in BAMPE mode which determines fragments size from proper paired read when it possible, otherwise fall backs to extending single end reads according to mean fragment size determined from paired reads ?

**Comment by taoliu** [comment]
Currently, there is no such feature in BAMPE mode to support both se and pe reads. My suggestion is that you can utilize the more flexible BEDPE support. You can convert your PE BAM to BEDPE format (not the one used by bedtools), so that the 1st, 2nd, 3rd columns tell the chromosome, start (leftmost end of the left read) and end (rightmost end of the right read) of the whole fragment; then convert your SE BAM to BEDPE format as well by extending them towards 3' ends with a fixed size, e.g.:

Your read in SE BAM is 'chr1 1000 1050 . . +', then if you want to extend it with 200bps, write 'chr1 1000  1200'. If your read in SE BAM is at - strand as 'chr1 1000 1050 . . -', then write 'chr1 850  1050'.

Put them together and use MACS2 callpeak with format as 'BEDPE'.
